_inc/lib/core-api/class.jetpack-core-api-module-endpoints.php

```diff
@@ -426,15 +430,24 @@ public function get_all_options() {
 					}
 					break;
 
+				case 'onboarding':
+					$response[ $setting ] = array(
+						'siteTitle' => get_option( 'blogname' ),
+						'siteDescription' => get_option( 'blogdescription' ),
+						'siteType' => get_option( 'jpo_site_type' ),
+						'homepageFormat' => get_option( 'jpo_homepage_format' ),
+						'addContactForm' => get_option( 'jpo_contact_page' ),
+						'businessAddress' => array(), // TODO
+						'installWooCommerce' => is_plugin_active( 'woocommerce/woocommerce.php' ),
+					);
+					break;
+
```

Comment:
This works fine; however, I wonder if the `onboarding` settings are really that much of a special case that they deserve to be handled here; as opposed to living inside [`Jetpack_Core_Json_Api_Endpoints::get_updateable_data_list()`](https://github.com/Automattic/jetpack/blob/4c52c67bffba463ae9531dc3fa8b378022cc784e/_inc/lib/class.core-rest-api-endpoints.php#L1228), which is used to [create the `$settings` array](https://github.com/Automattic/jetpack/pull/8497/files#diff-2ccec9069213650359171727de9a2471L406) over which we're iterating here.

Reply:
Since [`onboarding` is already part of `Jetpack_Core_Json_Api_Endpoints::get_updateable_data_list()`](https://github.com/Automattic/jetpack/blob/4c52c67bffba463ae9531dc3fa8b378022cc784e/_inc/lib/class.core-rest-api-endpoints.php#L1892-L1914), I wonder why https://github.com/Automattic/jetpack/pull/8497/files#diff-2ccec9069213650359171727de9a2471R446 isn't automatically exposing it for us?

Reply:
(It's currently simply returned as `false`, so maybe the schema and/or [`validate` callback](https://github.com/Automattic/jetpack/blob/4c52c67bffba463ae9531dc3fa8b378022cc784e/_inc/lib/class.core-rest-api-endpoints.php#L1912) needs tweaking.)

Reply:
It will not expose it, because the `jetpack_onboarding` option currently contains the onboarding token. In order for it to return all of the onboarding data, we need to transform it - which is what this PR is essentially doing.

Reply:
Right, there's no single option (as in `get_option()`) that contains all the data we need; rather, we have to aggregate them from different options 👍

Reply:
Exactly 👍